### ISSUE TITLE
DPO loss constant, logits chosen/rejected identical, and rewards nan

### ISSUE THREAD

**Issue by solume** [huggingface/trl#2553]
### System Info

- Platform: Linux-5.10.0-33-cloud-amd64-x86_64-with-glibc2.31
- Python version: 3.10.14
- PyTorch version: 2.4.0
- CUDA device(s): NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB
- Transformers version: 4.47.1
- Accelerate version: 1.2.1
- Accelerate config: not found
- Datasets version: 3.1.0
- HF Hub version: 0.24.6
- TRL version: 0.13.0
- bitsandbytes version: 0.39.0
- DeepSpeed version: 0.15.0
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.45.0
- PEFT version: 0.4.0

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

```python
import os
import torch

from transformers import AutoTokenizer, TrainingArguments, AutoModelForCausalLM
from datasets import load_dataset
from trl import DPOTrainer, DPOConfig

beta=0.9

output_dir= "/home/jupyter/dpo-out"
model_id = "meta-llama/Llama-3.2-3B-Instruct-sft"

dataset = load_dataset("json", data_files='dpo.jsonl',split="train")
dataset = dataset.shuffle(seed=42)
#dataset = dataset.select(range(1000))

training_args = DPOConfig(
    output_dir=output_dir,
    per_device_train_batch_size=8,
    gradient_accumulation_steps=2,
    logging_dir=output_dir+"/logs",
    learning_rate=1e-6,
    lr_scheduler_type="cosine",
    num_train_epochs=2,
    save_strategy="epoch",
    logging_steps=1,
    optim="paged_adamw_32bit",
    warmup_steps=2,
    max_prompt_length=1000,
    max_length=1000,
    bf16=True,
    fp16=False,
    tf32=True,
    beta=beta,
)

model = AutoModelForCausalLM.from_pretrained(model_id, trust_remote_code=True)
model.gradient_checkpointing_enable(gradient_checkpointing_kwargs={"use_reentrant":False})

model_ref = AutoModelForCausalLM.from_pretrained(model_id, trust_remote_code=True)
model_ref.gradient_checkpointing_enable(gradient_checkpointing_kwargs={"use_reentrant":False})

tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)
tokenizer.pad_token = tokenizer.eos_token
tokenizer.padding_side = "right"

def return_prompt_and_responses(samples):
    prompts = [x for x in samples["p"]]
    chosen = [x for x in samples["c"]]
    rejected = [x for x in samples["r"]]
    return {
        "prompt": prompts,
        "chosen": chosen,
        "rejected": rejected,
    }

original_columns = dataset.column_names

dataset = dataset.map(
    return_prompt_and_responses,
    batched=True,
    remove_columns=original_columns
)

print("loaded dataset & model")

dpo_trainer = DPOTrainer(
    model,
    model_ref,
    args=training_args,
    train_dataset=dataset,
    tokenizer=tokenizer,
)

dpo_trainer.train()
dpo_trainer.save_model(output_dir)

```

outputs:

```
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 5e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.06}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 1e-06, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.12}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 9.972609476841365e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.19}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 9.890738003669027e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.25}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 9.755282581475767e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.31}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 9.567727288213004e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.38}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 9.330127018922193e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.44}
{'loss': 0.6914, 'grad_norm': 0.0, 'learning_rate': 9.045084971874737e-07, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': 0.0, 'logps/rejected': 0.0, 'logits/chosen': nan, 'logits/rejected': nan, 'epoch': 0.5}
```

### Expected behavior

I would expect the loss to decrease, the logits to not be identical values, and the rewards to not be nan.

### Checklist

- [X] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [X] I have included my system information
- [X] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [X] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [X] Any traceback provided is complete

**Comment by edbeeching** [comment]
This is probably a dataset issue, can you test your script with a standard DPO dataset such as `trl-lib/ultrafeedback_binarized` to ensure there are no NaNs?

**Comment by AIR-hl** [comment]
same problem
